stripe/src/main/java/com/stripe/android/view/CardNumberEditText.java

```diff
@@ -185,13 +193,17 @@ public void afterTextChanged(Editable s) {
         });
     }
 
-    private void updateCardBrand(String partialNumber) {
-        @Card.CardBrand String oldBrand = mCardBrand;
-        mCardBrand = CardUtils.getPossibleCardType(partialNumber);
-        if (mCardBrand.equals(oldBrand)) {
+    private void updateCardBrand(@NonNull @Card.CardBrand String brand) {
```

Comment:
It's pre-existing, but intentional due to the preference not to use Enums in Android because of dexlimits and object size. CardBrand is actually getting pretty close to where it would be worth making an Enum out of it rather than a @StringDef. I'd be open to doing that in the future.